sky/backends/backend_utils.py

```diff
@@ -523,34 +522,27 @@ def add_cluster(
                 break
 
         proxy_command = auth_config.get('ssh_proxy_command', None)
-        docker_proxy_command = None
-        head_port = ports[0]
+
+        docker_proxy_command_generator = None
         if docker_user is not None:
             docker_proxy_command_generator = lambda ip: ' '.join(
```

Comment:
nit: Can we add the port to this function as well as we mentioned in https://github.com/skypilot-org/skypilot/pull/2706#discussion_r1435827830? In the future, some of other clouds do not use the default port for the VM.